client/me/purchases/confirm-cancel-domain/index.jsx

```diff
@@ -0,0 +1,243 @@
+			return;
+		}
+
+		return (
+			<div className="confirm-cancel-domain__help_message">
+				<p>
+					{ selectedReason.helpMessage }
+				</p>
+				{ selectedReason.showTextarea &&
+				<FormTextarea onChange={ this.onMessageChange } rows="5" cols="60" /> }
+			</div>
+		);
+	},
+
+	renderConfirmationCheckbox() {
+		if ( ! this.isValidReasonToCancel() ) {
+			return;
+		}
+
+		return (
+			<div className="confirm-cancel-domain__confirm_container">
+				<FormLabel>
+					<FormCheckbox name="confirm"
+							checked={ this.state.confirmed }
+							onChange={ this.onConfirmationChange } />
+					<span>
+						{ this.translate(
+							'I understand that canceling means that I may {{strong}}lose this domain forever{{/strong}}.', {
+								components: {
+									strong: <strong />
+								}
+							}
+						) }
+					</span>
+				</FormLabel>
+			</div>
+		);
+	},
+
+	renderSubmitButton() {
+		if ( ! this.isValidReasonToCancel() ) {
+			return;
+		}
+
+		if ( this.state.submitting ) {
+			return (
+				<FormButton disabled={ true } >
+					{ this.translate( 'Cancelling Domain…' ) }
+				</FormButton>
+			);
+		}
+
+		const selectedReason = this.state.selectedReason,
+			confirmed = this.state.confirmed;
+
+		if ( selectedReason && 'misspelled' === selectedReason.value ) {
+			return (
+				<FormButton onClick={ this.onSubmit } disabled={ ! confirmed } >
+					{ this.translate( 'Cancel Anyway' ) }
+				</FormButton>
+			);
+		}
+
+		return (
+			<FormButton onClick={ this.onSubmit } disabled={ ! confirmed } >
+				{ this.translate( 'Cancel Domain' ) }
+			</FormButton>
+		);
+	},
+
+	render() {
+		if ( this.isDataLoading() ) {
+			return <ConfirmCancelDomainLoadingPlaceholder
+				purchaseId={ this.props.purchaseId }
+				selectedSite={ this.props.selectedSite } />;
+		}
+
+		const purchase = getPurchase( this.props );
+
+		if ( ! isDomainRegistration( purchase ) ) {
+			return;
+		}
+
+		const domain = getDomainName( purchase ),
+			selectedReason = this.state.selectedReason;
+
+		return (
+			<Main className="confirm-cancel-domain">
+				<HeaderCake onClick={ goToCancelPurchase.bind( null, this.props ) }>
+					{ titles.confirmCancelDomain }
+				</HeaderCake>
+				<Card>
+					<h3>{ this.translate( 'Canceling %(domain)s', { args: { domain } } ) }</h3>
+					<p>
+						{ this.translate(
+							'Since domain cancellation can cause your site to stop working, ' +
+							'we’d like to make sure we help you take the right action. ' +
+							'Please select the best option below.'
+						) }
+					</p>
+					<SelectDropdown
+							className="confirm-cancel-domain__reasons_dropdown"
+							key="confirm-cancel-domain__reasons_dropdown"
+							selectedText={ selectedReason
+											? selectedReason.label
+											: this.translate( 'Please let us know why you wish to cancel.' ) }
+							options={ cancellationReasons }
+							onSelect={ this.onReasonChange } >
+					</SelectDropdown>
+					{ this.renderHelpDialog() }
```

Comment:
I'm not sure, it's a paragraph of text (helping the user depending on the reason of his cancelation) and eventually a `textarea` input. Do you have a better idea for the name of this function?